platforms/core-runtime/base-services/src/main/java/org/gradle/internal/jvm/Jvm.java

```diff
@@ -364,7 +365,7 @@ private Optional<File> findToolsJar(File javaHome) {
     public boolean isIbmJvm() {
         for (String vendorProperty : VENDOR_PROPERTIES) {
             if (System.getProperties().containsKey(vendorProperty)
-                && System.getProperty(vendorProperty).toLowerCase().startsWith("ibm corporation")) {
+                && System.getProperty(vendorProperty).toLowerCase(Locale.ROOT).startsWith("ibm corporation")) {
```

Comment:
Two reasons against `TextUtil.toLowerCaseLocaleSafe()`:

- It adds a dependency on that class, and that class is currently in `:base-services` (yes, this class is there, too, but I hope not for long); for a trivial piece of code I'd like to avoid doing this.
- That method uses `Locale.ENGLISH`, which could be good for this one, though I don't think there's a specification about what language `java.vendor` should be in. So I think `ROOT` is the best common denominator here.

There are so many ways we could fix this, too.

- We could compare against `IBM Corporation`, because that's what the vendor will be set to. There's no way IBM will use anything but all-uppercase here.
- We could drop the whole method, since we don't have any code that actually depends on an IBM JVM anymore. IBM JDK 1.6 had some quirks to it on Windows IIRC, but since then these problems are gone. It also looks like IBM stopped delivering their own-branded SDKs with JDK 8: https://www.ibm.com/support/pages/java-sdk-downloads.

I'd just leave this as it is, and maybe remove it once we drop support for Java 8 entirely.

Reply:
Note that ErrorProne recommends `ROOT` or `ENGLISH` in https://errorprone.info/bugpattern/StringCaseLocaleUsage. I think in general `ROOT` is better for non-user-input like enum values, as they are not really "English" words.